docs/addon-guide.md

```diff
@@ -0,0 +1,193 @@
+Addons
+
+An addon is an extension of the Hub and consists of a container (image) and a
+manifest (Custom Resource). The CR defines the addon and registers it with the Hub.
+It is intended to provide business logic that contributes to the application 
+*knowledge-base* (inventory) or produces modernization assets. An addon is essentially
+just a binary written in any language that is integrated through the Hub API. The addon
+must report its status using the task (addon) report:
+1. Report addon started: `POST /tasks/:id/report` Task.State=Running.
+2. Report addon activity & progress: `PUT /tasks/:id/report` (repeat as needed).
+3. Report addon completed: `PUT /tasks/:id/report` Task.State=(Succeeded|Failed)
+
+The knowledge about an application consists of:
+- Application (model).
+- Source repository.
+- Tags.
+- Application Facts (key/value pair).
+- Files (stored in the application *bucket*).
+- Assessments.
+
+Modernization assets may be stored in:
+- Source repository.
+- Application Facts (key/value pair).
+- Files (stored in the application *bucket*).
+
+## Definition
+
+An Addon is defined and registered with the Hub using a Custom Resource (CR).
+
+* **image** - The addon image.
+* **resources** - An *(optional)* standard k8s pod container [resources](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers) specification.
+* **mounts** - An optional list of volumes to be mounted in the addon pod.
+    * **claim** - PVC name.
+    * **name** - The name of the directory in `/mnt` to mount the volume.
+
+Example:
+```
+apiVersion: tackle.konveyor.io/v1alpha1
+kind: Addon
+metadata:
+  name: admin
+  namespace: konveyor-tackle
+spec:
+  image: quay.io/konveyor/tackle2-addon:latest
+  mounts:
+  - claim: tackle-maven-volume-claim
+    name: m2
+  resources:
+    requests:
+      cpu: 50m
+      memory: 50Mi
+```
+
+## Runtime (pod)
+
+### Environment
+
+The pod is started with the following environment variables:
+  - **ADDON_WORKING_DIR** - Path to addon working directory.
+  - **HUB_BASE_URL** - Base URL for the hub API.
+  - **TOKEN** - An authentication token.
+  - **TASK** - An associated task ID.
+
+The bucket volume is mounted in `/buckets`.
+
+The addon container must have the default entry point set.
```

Comment:
I don't think it's required if the pod spec includes a _command_ (and _args_) section which I'd hoped to avoid. So, this is explicitly telling the writer a default entry point is required.
We can discuss the need for addon configuration beyond what is provided but initially hoped to avoid it.